### PR TITLE
[EDI] UDES-14 - Gateway menu is slow when lots of transfers

### PR DESCRIPTION
**Pull request by vvasoyav**
EDI gateway menu is slow down when lots of edi transfers happens. so it is resolved by improve the manage transfer.

Story :: SE-18

Signed off By :: vishal.vasoya@unipart.com

### CODE REVIEW

File: addons/edi/models/edi_gateway.py [hunk 1]
```diff
@@ -193,8 +193,17 @@ def _compute_path_count(self):
...
         """Compute number of transfers (for UI display)"""
+
+        #We use the read_group method to aggregate data from the edi.transfer model. 
+        #We specify the search domain to filter transfers related to the current edi.gateway, 
+        #group the results by gateway_id, and count the number of records for each group.
+    
+        transfers_data = self.env['edi.transfer'].read_group(
+        [('gateway_id', 'in', self.ids)],
+        ['gateway_id'], ['gateway_id'])
+        transfer_count_dict = {data['gateway_id'][0]: data['gateway_id_count'] for data in transfers_data}
```
Comment (rajeeshknambiar) [inline addons/edi/models/edi_gateway.py]: Nit: black formatting

```py
        transfers_data = self.env["edi.transfer"].read_group(
            [("gateway_id", "in", self.ids)], ["gateway_id"], ["gateway_id"]
        )
        transfer_count_dict = {
            data["gateway_id"][0]: data["gateway_id_count"] for data in transfers_data
        }
```

Comment (vvasoyav) [reply addons/edi/models/edi_gateway.py]: Yeah Changed.

Comment (armandcela) [reply addons/edi/models/edi_gateway.py]: I see you pushed a new commit, but don't see that pushed suggested changes. Still black formatting is not applied and the other suggestions from 11 MR-s which they are same for this MR as well. Am i missing smth?

Comment (vvasoyav) [reply addons/edi/models/edi_gateway.py]: I have applied chanes in botn MRs can u please check it again and in 14 only one commit is there but for 11 there 2 commits.

**Review by armandcela — APPROVED** [review]

**Review by rajeeshknambiar — APPROVED** [review]

**Pull request merged**